math/src/main/java/com/powsybl/math/timeseries/TimeSeries.java

```diff
@@ -0,0 +1,209 @@
+/**
+ * Copyright (c) 2017, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.math.timeseries;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.powsybl.commons.json.JsonUtil;
+import gnu.trove.list.array.TDoubleArrayList;
+import gnu.trove.list.array.TIntArrayList;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.io.Writer;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Stream;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public interface TimeSeries<P extends AbstractPoint, C extends ArrayChunk<P>> extends Iterable<P> {
+
+    TimeSeriesMetadata getMetadata();
+
+    List<C> getChunks();
+
+    Stream<P> stream();
+
+    Iterator<P> iterator();
+
+    void writeJson(JsonGenerator generator);
+
+    static void writeJson(JsonGenerator generator, List<TimeSeries> timeSeriesList) {
```

Comment:
Assert generator and timeSeriesList are not null

Reply:
Done